README.md

```diff
@@ -29,117 +29,354 @@ The project goal is to create a simple marketplace for book lovers, while not ha
+        "bookCondition": "New",
+        "price": 999.99
+    }
+    ```
+
+    ### Updating a Book
+
+    - **PUT** `http://localhost:8080/books/update/{id}`
+
+    ```json
+    {
+        "title": "Everything you dont need to know about software engineering",
+        "isbn": 11111111,
+        "publishDate": "2020-02-20",
+        "author": "Christopher Hong",
+        "genre": "Cool",
+        "bookCondition": "Old",
+        "price": 999999.99
+    }
+    ```
+
+    ### Counting Books
+
+    - **GET** `http://localhost:8080/books/count`
+
+    ### Getting a Book by ID
+
+    - **GET** `http://localhost:8080/books/{id}`
+
+    ### Deleting a Book by ID
+
+    - **DELETE** `http://localhost:8080/books/delete/{id}`
+
+    Make sure that the `bookId` is not in the `book_listings` table before deletion.
+
+    ## Listing Operations
+
+    ### Creating a New Listing
+
+    - **POST** `http://localhost:8080/listings/create`
+
+    ```json
+    {
+        "bookId": 1,
+        "userId": 1,
+        "listingStatus": "new"
+    }
+    ```
+
+    ### Updating a Listing
+
+    - **PUT** `http://localhost:8080/listings/update/{id}`
+
+    Make sure the `bookId` and `userId` refer to existing records.
+
+    ```json
+    {
+        "bookId": 2,
+        "userId": 2,
+        "listingStatus": "old"
+    }
+    ```
+
+    ### Counting Listings
+
+    - **GET** `http://localhost:8080/listings/count`
+
+    ### Getting a Listing by ID
+
+    - **GET** `http://localhost:8080/listings/{id}`
+
+    ### Deleting a Listing by ID
+
+    - **DELETE** `http://localhost:8080/listings/delete/{id}`
+
+    ## Book Tag Operations
+
+    ### Creating a New Book Tag
+
+    - **POST** `http://localhost:8080/booktags/create`
+
+    ```json
+    {
+        "tagName": "Science Fiction",
+        "bookId": 1
+    }
+    ```
+
+    ### Getting a Book Tag by ID
+
+    - **GET** `http://localhost:8080/booktags/{id}`
+
+    ### Getting Book Tags Count
+
+    - **GET** `http://localhost:8080/booktags/count`
+
+    ### Updating a Book Tag
+
+    - **PUT** `http://localhost:8080/booktags/update/{id}`
+
+    ```json
+    {
+        "tagName": "Fantasy",
+        "bookId": 2
+    }
+    ```
+
+    ### Deleting a Book Tag by ID
+
+    - **DELETE** `http://localhost:8080/booktags/delete/{id}`
+
+    ## Book Transaction Operations
+
+    Make sure that the referenced `buyerId` and `sellerId` exist in the `users` table (userIds), and the `listingId` exists in the `book_listings` table.
+
+    ### Creating a New Book Transaction
+
+    - **POST** `http://localhost:8080/transactions/create`
+
+    Make sure the `buyerId`, `sellerId`, and `listingId` refer to existing records.
+
+    ```json
+    {
+        "buyerId": 1,
+        "sellerId": 2,
+        "listingId": 1,
+        "transactionPrice": 250.00,
+        "transactionStatus": "completed"
+    }
+    ```
+
+    ### Getting a Book Transaction by ID
+
+    - **GET** `http://localhost:8080/transactions/{id}`
```

Comment:
change transactions/ to book-transactions/

Reply:
oops